src/RadixDlt.NetworkGateway.PostgresIntegration/Services/TransactionQuerier.cs

```diff
@@ -157,47 +148,52 @@ public async Task<ICollection<StatusLookupResult>> LookupPendingTransactionsByIn
         return pendingTransactions.Select(pt => new StatusLookupResult(pt.PayloadHash.ToHex(), pt.Status.ToGatewayModel(), pt.FailureReason)).ToArray();
     }
 
-    private async Task<List<long>> GetRecentUserTransactionStateVersions(
-        RecentTransactionPageRequest request,
-        GatewayModel.LedgerState atLedgerState,
-        GatewayModel.LedgerState? fromLedgerState,
-        CancellationToken token)
+    private async Task<List<long>> GetTransactionStreamPageStateVersions(TransactionStreamPageRequest request, GatewayModel.LedgerState atLedgerState, CancellationToken token)
     {
-        if (fromLedgerState != null)
-        {
-            var bottomStateVersionBoundary = request.Cursor?.StateVersionBoundary ?? fromLedgerState.StateVersion;
-            var topStateVersionBoundary = atLedgerState.StateVersion;
+        var stateVersionBoundary = request.Cursor?.StateVersionBoundary ?? request.FromStateVersion;
 
-            return await _dbContext.LedgerTransactions
-                .Where(lt => lt.StateVersion >= bottomStateVersionBoundary && lt.StateVersion <= topStateVersionBoundary)
-                .OrderBy(at => at.StateVersion)
-                .Take(request.PageSize + 1)
-                .Select(at => at.StateVersion)
-                .ToListAsync(token);
+        var query = _dbContext.LedgerTransactions
+            .Where(lt => lt.StateVersion <= atLedgerState.StateVersion);
+
+        if (request.AscendingOrder)
+        {
+            query = query.Where(lt => lt.StateVersion >= stateVersionBoundary)
```

Comment:
Because with ascending order `next_cursor` is going to point to the higher state version while with regular descending order it is going to be lower one.

Reply:
ok so I as susspected.